pkg/virt-handler/vm.go

```diff
@@ -1272,9 +1272,10 @@ func (d *VirtualMachineController) checkNetworkInterfacesForMigration(vmi *v1.Vi
 		networks[network.Name] = network.DeepCopy()
 	}
 	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		if iface.Bridge != nil && networks[iface.Name].Pod != nil {
-			return fmt.Errorf("cannot migrate VMI with a bridge interface connected to a pod network")
+		if iface.Masquerade != nil || iface.Bridge != nil {
+			continue
 		}
+		return fmt.Errorf("cannot migrate VMI with a bridge interface connected to a pod network")
```

Comment:
This does not match the logic. If there is a bridge interface, it is skipped `continue` in the previous block. Also, you don't check for the network type `Pod`, so this would be raised for secondary interfaces too.